SAP2000_Adapter/CRUD/Read/Loads.cs

```diff
@@ -520,18 +526,75 @@ private List<ILoad> ReadBarUniformTemperatureLoad(List<string> id = null)
+                    if (loadType[i] == 1)
+                    {
+                        loads.Add(Engine.Structure.Create.BarUniformTemperatureLoad(bhomCases[caseNames[i]], tempChange, new BHoMGroup<Bar> { Elements = { bhomBar } }, LoadAxis.Global, false));
+                    }
+                }
+            }
+
+            return loads;
+        }
+
+        /***************************************************/
+
+        private List<ILoad> ReadBarDifferentialTemperatureLoad(List<string> id = null)
+        {
+            List<ILoad> loads = new List<ILoad>();
+
+            Dictionary<string, Loadcase> bhomCases = ReadLoadcase().ToDictionary(x => x.Name.ToString());
+            Dictionary<string, Bar> bhomBars = ReadBars().ToDictionary(x => GetAdapterId<string>(x));
+
+            int count = 0;
+            string[] barNames = null;
+            string[] caseNames = null;
+            int[] loadType = null;
+            string[] jointPattern = null;
+            double[] val = null;
+
+            if (m_model.FrameObj.GetLoadTemperature("ALL", ref count, ref barNames, ref caseNames, ref loadType, ref val, ref jointPattern, eItemType.Group) == 0)
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    Bar bhomBar = bhomBars[barNames[i]];
 
-                    loads.Add(new BarUniformTemperatureLoad()
+                    if (jointPattern[i] == "None")
                     {
-                        TemperatureChange = tempChange,
-                        Loadcase = bhomCases[caseNames[i]],
-                        Objects = new BHoMGroup<Bar> { Elements = { bhomBar } },
-                        Axis = LoadAxis.Global,
-                        Projected = false,
-                    });
+                        Engine.Reflection.Compute.RecordWarning("BarDifferentialTemperatureLoads cannot vary along the bar - only constant temperatures and variation across the bar are supported. The load has been ignored.");
+                    }
+                    else
+                    {
+                        switch (loadType[i])
+                        {
+                            case 2:
+                                double width = bhomBar.SectionProperty.Vpy + bhomBar.SectionProperty.Vy;
+                                loads.Add(new BarDifferentialTemperatureLoad()
+                                {
+                                    TemperatureProfile = new Dictionary<double, double>() { { 0, 0 }, { 1, val[i] * width } },
+                                    LoadDirection = DifferentialTemperatureLoadDirection.LocalY,
+                                    Loadcase = bhomCases[caseNames[i]],
+                                    Objects = new BHoMGroup<Bar> { Elements = { bhomBar } },
+                                    Axis = LoadAxis.Global,
+                                    Projected = false,
+                                });
```

Comment:
Was there not an earlier commit that made use of the Engine methods?

Reply:
hmm did I do something weird? lemme check.

Reply:
ah, took it out of one method, not out of this. I'm raising a new issue to use engine methods for all the reads, #246, and changing this method.